apisix/core/config_etcd.lua

```diff
@@ -123,6 +124,15 @@ local function waitdir(etcd_cli, key, modified_index, timeout)
         res, err = res_func()
     end
 
+    if http_cli then
+        local res_cancel, err_cancel = etcd_cli:watchcancel(http_cli)
+        if res_cancel and res_cancel == 1 then
```

Comment:
`if res_cancel == 1 then` is enough

Reply:
Thx! Should I create a PR for it or leave it to the next PR

Reply:
sure,welcome